cmd/fs-v1-multipart.go

```diff
@@ -19,16 +19,61 @@ package cmd
 import (
 	"crypto/md5"
 	"encoding/hex"
+	"encoding/json"
 	"fmt"
 	"hash"
 	"io"
-	"path"
+	"io/ioutil"
+	"os"
+	pathutil "path"
 	"strings"
 	"time"
 
 	"github.com/minio/sha256-simd"
 )
 
+// listMultipartUploadIDs - list all the upload ids from a marker up to 'count'.
+func (fs fsObjects) listMultipartUploadIDs(bucketName, objectName, uploadIDMarker string, count int) ([]uploadMetadata, bool, error) {
+	var uploads []uploadMetadata
+
```

Comment:
now since LOCK_SH flock is held (rlk) what can happen is, a competing LOCK_EX in the same minio process will succeed after which if rlk.Close() happens then LOCK_EX will be lost. hence another minio server can hold LOCK_EX but we think we have LOCK_EX. i.e two processes can think that they have LOCK_EX at the same time.

previously there was no scope for this situation to arise.